### ISSUE TITLE
Feature Request: Clean up UIA Tree Navigation

### ISSUE THREAD

**Issue by carlos-zamora** [microsoft/terminal#2102]
# Summary of the new feature/enhancement

As of now, the UIA Tree is essentially what we get from XAML. It's great that we actually have a tree that we can navigate now, but now we need to clean it up a bit.

ConHost tends to shift the focus to the text buffer by default. We need to decide on a story for two things:
- how/when XAML's focus should be shifted (and to what)
- What UIA elements should be directly accessible (parent and children) from each UIA element

The implementation of the first doesn't require any knowledge of accessibility. You just need to add tab stops and do a bit of basic XAML changes.

The implementation of the second requires overriding/defining the Navigate() accessibility functions in our current UIA providers and XAML elements.

You can _technically_ do these separately. But when you change one, it'll make it a pain to test when navigating the UIA tree. Let's put our thoughts on how this should look like here!
